app/ide-desktop/lib/content/tailwind.config.ts

```diff
@@ -1,7 +1,7 @@
 /** @file Configuration for Tailwind. */
 /** @type {import('tailwindcss').Config} */
 module.exports = {
-    content: ['./src/**/*.tsx'],
+    content: ['../dashboard/src/authentication/src/**/*.tsx'],
```

Comment:
That's a very strange dependency here. We have `content` which does not use tailwind (is that correct?) and it refers another library `dashboard` which uses tailwind to build the styles? If Im correct, then the building of tailwind should be integrated in the build script of `dashboard` maybe? Or maybe this should be something more top-level? I don't understand why `content` is now responsible of building artifacts for other libraries. CC @PabloBuchu  and @mwu-tow - how should it be done in the correct way, do you have some thoughts here?

Reply:
i believe currently `content` contains the desktop IDE frontend, whereas `dashboard` contains the dashboard, which will be used for both cloud and desktop.

(so, if the dashboard is supposed to be in the desktop IDE, then it should be here too)

on second thought, this might be a bad solution. because the css is no longer in the frontend, and that might be an issue...

Reply:
maybe now is a good time to figure out how to *properly* structure `content` and `dashboard`

Reply:
> i believe currently `content` contains the desktop IDE frontend, whereas `dashboard` contains the dashboard, which will be used for both cloud and desktop.
> 
> (so, if the dashboard is supposed to be in the desktop IDE, then it should be here too)
> 
> on second thought, this might be a bad solution. because the css is no longer in the frontend, and that might be an issue...

That might be correct but this is VERY temporary solution. Content is meant to be run both in Electron and in the cloud. It is not being used in cloud only because there was no time to refactor it, but it should change ASAP, as now it is in a very confusing state. Dashboard is thus dependency of `content`, as it is displayed as part of `content`. `Dashboard` is a separate library, in a separate folder, next to `content`, and `content` imports it, which is IMO correct.

Reply:
about merging the code - maybe we could consider using vite to replace esbuild + create-react-app
vite:
- *does not* support esbuild plugins because the architecture is incompatible with their own plugins - so we'll have to look for alternatives. this shouldn't be too hard because vite has a pretty large ecosystem
- supports HMR (we are using this to speed up testing the cloud frontend)
- supports tailwind
- seems to support electron

importantly that means we could build the dashboard using the same build tool as content, which should make it a lot easier for content to depend on dashboard

Reply:
We've chosen `rebuild` because of one reason - fast build times. It is crucial for us to maintain as small build time as possible after changing Rust sources. Do you know how does vite compare to it?

Reply:
vite uses esbuild for transpilation, but it uses rollup fpr bundling which is written in js so it's not the fastest.

as an alternative to vite we could look into manually implementing some HMR support in development mode, or just not using HMR completely? because esbuild *does* support live reloads out of the box, it's just HMR that it doesnt

the downside of live reloading as compared to HMR is that the cloud frontend would send requests to the backend every time the page is live-reloaded, but that may not be much of a problem for us

Reply:
note that not using HMR should barely affect desktop IDE, since it is not sending requests to the backend (no network requests = latency for e.g. listing projects is a lot lower)

Reply:
Could you chat about this with @mwu-tow, please? He was setting up `esbuild` here and he has the best knowledge of how the build system works / should work.

Reply:
@wdanilo This is a problem with single `index.html` and tailwind.css dependency. Styles are being used in `/dashboard` app but `tailwind` itself (and config) is being built in content since it is built by esbuilt. 

In dashboard we don't have any place / building scripts to do that since it is the content that also runs dashboard/authentication

Happy to jump in on call and quick chat on this since this PR fixes for us an important bug

Reply:
So to answer — I am not knowledgeable about web development nor the whole js, ts, react ecosystems. `esbuild` was chosen because it was much faster than `webpack` (which we used previously) and covered our needs. IIRC, at the time of that decision, we hadn't expected that any subproject that requires anything not supported by `esbuild` would be added. The dashboard was planned to be written in Rust.

Are there other problems with using `esbuild` for `dashboard` apart from HMR? Why was `create-react-app` that uses `webpack` under the hood introduced after we decided to replace all `webpack` with `esbuild`? Was there an explicit decision for this choice, or was it just a default that persisted?

My quick thoughts:
* If both `content` and `dashboard` require `tailwind.css`, then both of them should have it as their dependency, and their build systems should support it in their respective ways.
* We should not care about deduplicating this tailwind generation by ourselves; this should be something that build system (being aware of the whole dependency tree) should handle. In the worst-case scenario, setting the file twice is not a disaster.
* `content` should not assume that `dashboard` uses tailwind; it is an implementation detail.
* It would be good to switch to `esbuild` for `dashboard` without too much effort.
* If something like `vite` can replace `esbuild` for all our projects and save us some manual tinkering over `esbuild`, then it is even better, but I'm concerned about the necessary work to migrate to it.

Reply:
> are there any other problems?
i don't think so. if there are any, i think they will be minor enough that they can and should be solved now rather than later

> why was `create-react-app` used even though it uses `webpack`?
i assume it's just most people's go-to template - so yes, a default that persisted

> If both `content` and `dashboard` require `tailwind.css`
well, `content` requires/will require `dashboard` which in turn requires `tailwind.css`

i think i'll try switching to esbuild for `dashboard`. if that works i'll probably open up a separate PR

Reply:
@wdanilo @mwu-tow can we transform this path issue into a FIXME? We will refactor how the dashboard and content is handled after we clean up the rest of the merges.

Reply:
I agree 100% with what @mwu-tow has written. Btw, Michal, thanks for breaking that down so clearly, as always ❤️ 

Anyway, if content do not uses tailwind, and probably will never use it, the tailwind building should not be integrated with content, it should be integrated with `dashboard`. I think the idea to move dashboard to `esbuild` and keep every part of our TS/JS ecosystem using the same build tools is the right choice @indiv0. Then, we can just integrate building tailwind in the `dashboard` `esbuild` config, and use `dashboard` as a library in `content`, while keeping `content` unaware of internal implementation details of `dashboard`. 

@PabloBuchu I'm OK with merging this as is iff we will be fixing that in the following PRs. This introduces pretty bad dependency between packages and makes the tech debt much worse. So yes, we can move forward, but let's please fix it as @mwu-tow, @indiv0, and @somebody1234 proposed above :)